File: backend/src/schemas/file.py
"""
ENHANCEMENT L2: FILE ATTACHMENTS

Pydantic schemas for file upload, download, and attachment API responses
Defines the data structures used for file-related API operations
"""

from datetime import datetime
from typing import List
from pydantic import BaseModel, Field


class FileUploadResponse(BaseModel):
    """
    ENHANCEMENT L2: FILE ATTACHMENTS - Response schema for file upload
    """
    id: str
    filename: str
    content_type: str
    size: int
    url: str
    uploaded_at: datetime


class AttachFilesRequest(BaseModel):
    """
    ENHANCEMENT L2: FILE ATTACHMENTS - Request schema for attaching files to a ticket
    """
    file_ids: List[str] = Field(..., description="List of file IDs to attach")


class FileAttachmentResponse(BaseModel):
    """
    ENHANCEMENT L2: FILE ATTACHMENTS - Response schema for file attachment
    """
    id: str
    filename: str
    content_type: str
    size: int
    url: str
    uploaded_at: datetime

File: frontend/src/app/shared/types/file.ts
/**
 * Shared file attachment types
 */

// Base file attachment from API
export interface FileAttachment {
  id: string;
  filename: string;
  content_type: string;
  size: number;
  uploaded_at: string;
  url?: string;
}

// Extended file attachment for upload component
export interface FileAttachmentUpload extends Omit<FileAttachment, 'id' | 'filename' | 'content_type' | 'uploaded_at'> {
  id?: string;
  file?: File;
  name: string;
  type: string;
  uploadProgress?: number;
  uploadError?: string;
  uploaded?: boolean;
}

// File upload response from API
export interface FileUploadResponse {
  id: string;
  filename: string;
  content_type: string;
  size: number;
  url: string;
  uploaded_at: string;
}